profiling/src/logging.rs

```diff
@@ -0,0 +1,20 @@
+use env_logger::Target;
+pub use log::Level;
+pub use log::LevelFilter;
+use std::fs::File;
+use std::os::unix::io::FromRawFd;
+
+pub fn log_init(level_filter: LevelFilter) {
+    // Safety: this is safe, it's just "unsafe" because it's a call into C.
+    let fd = unsafe { libc::dup(libc::STDERR_FILENO) };
+    if fd != -1 {
+        // Safety: the fd is a valid and open file descriptor, and the File has sole ownership.
+        let target = Box::new(unsafe { File::from_raw_fd(fd) });
+        env_logger::builder()
+            .filter_level(LevelFilter::Off)
+            .filter_module("datadog_php_profiling", level_filter)
+            .target(Target::Pipe(target))
```

Comment:
Is there a reason to do our own fd-duplicating logic, or can we use `Target::Stderr`?

Reply:
Yes, there is! I will document this. For almost all released versions the PHP CLI will close the stderr stream between `rshutdown` and `prshutdown`, but there are still other things that write to stderr after this (not just in this extension, but in other extensions too). Certain Rust libs are definitely _not_ expecting that to happen, and I was getting panics. That's actually why I am using `env_logger`: because it supports using a target based on a file descriptor.
I fixed this in PHP for the latest patch releases of PHP 8, but I still need to support PHP 7, and it will take time for these fixes to roll out everywhere.